File: 0077-combinations/0077-combinations.java
class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<Integer> al=new ArrayList<>();
        List<List<Integer>> ans=new ArrayList<>();
        int[] a=new int[n];
        int counter=1;
        for(int i=0;i<a.length;i++)
            a[i]=counter++;
        findComb(a,0,k,al,ans);
        return ans;
    }

    static void findComb(int[] a,int idx,int size,List<Integer> al,
    List<List<Integer>> ans)
    {
        if(al.size()==size){
            ans.add(new ArrayList<Integer>(al) );
            return;
        }

        for(int i=idx;i<a.length;i++){
            al.add(a[i]);
            findComb(a,i+1,size,al,ans);
            al.remove(al.size()-1);
        }
    }
}

File: 0077-combinations/0077-combinations.py
class Solution(object):
    def combine(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: List[List[int]]
        """
        sol=[]
        ans=[]
        self.get_combinations(n+1,k,1,ans,sol)
        return sol
        

    def get_combinations(self,n,k,idx,ans,sol):
        if(idx>n):
            return
        elif(k==0):
            b=[]
            for i in range(len(ans)):
                b.append(ans[i])
            sol.append( b)
        else:
            ans.append(idx)
            self.get_combinations(n,k-1,idx+1,ans,sol)
            ans.remove(idx)
            self.get_combinations(n,k,idx+1,ans,sol)